src/hash.hpp

```diff
@@ -5,10 +5,22 @@
 #include <string>
 #include <vector>
 
+#include "util/result.hpp"
+
 namespace fur::hash {
+
 /// Type for a SHA1 hash
 using hash_t = std::array<uint8_t, 20>;
 
+enum class HashError{
+  // The hash of the given string is malformed
+  MalformedPieceHashesString
+};
+
+/// Result of a hash operation
+typedef util::Result<std::vector<hash_t>, HashError>
```

Comment:
This is C syntax, should prefer `using`.

Reply:
I also prefer `using`, now I will change all Results